fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java

```diff
@@ -90,7 +90,7 @@
     // The fifth group for allows ACL.
     // The sixth group allows for any hashed suffixes on the acls.
     private final static Pattern FORWARD_COMPONENT_PATTERN = Pattern.compile(
-            "(.*?)(/" + FCR_METADATA + ")?(/" + FCR_VERSIONS + "(/\\d+)?)?(/" + FCR_ACL + "(\\#\\w+)?)?$");
+            "(.*?)(/" + FCR_METADATA + ")?(/" + FCR_VERSIONS + "(/.+)?)?(/" + FCR_ACL + "(\\#\\w+)?)?$");
```

Comment:
I am reluctant to change this regex if the reason is just for the integration test.

Reply:
@awoods I think one of the reasons for the pattern change is @whikloj's `19/Jul/18 1:15 PM` comment https://jira.duraspace.org/browse/FCREPO-2821#comment-60168: `We do (at least I think) need to alter the FORWARD_COMPONENT_PATTERN as that would make GET requests to things like /fcr:versions/something fail with a 404 instead of a 400.` 

And I am also seeing that with the origin pattern to allow digits only for Memento ID, if an incoming path is something like http://localhost:8080/rest/test4/obj/fcr:versions/new, it won't get translated to the internal path `/test4/obj/fedora:Timemap/new`, and I need to do a special check to see whether it can be translated or not before testing for the existing of the JCR node, which will ensure status 405 return. See ContentExposingResource.java#L574 (https://github.com/fcrepo4/fcrepo4/pull/1375/commits/44b76a9d8c05c669fa149bc37d767506f415cc23#diff-d372035f89c8d0846920e20bd13e0145R574).

Reply:
My understanding is that memento IDs are being generated by Fedora... that we do not allow users to specify their own names for mementos, and therefore the regex should never be violated.
Can you confirm @bbpennel ?

Reply:
@awoods / @bbpennel Do you want me to change it back to use the digits pattern for memento ID? I think it's correct that the memento IDs are being generated by FedoraI at this time and it's always digits. But for an incoming path like `/rest/test4/obj/fcr:versions/new`, it won't get converted properly to the corresponding internal path `/rest/test4/obj/fedora:timemap/new` when using the digits pattern.

Reply:
> but it may be best to separate that from the test to verify that PUT/POST/PATCH are being disallowed, and that a resources doesn't exist.

👍 👍